api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -138,6 +138,14 @@ static StructType getType(StructType partitionType) {
    */
   DataFile copy();
 
+  /**
+   * Copies this {@link DataFile data file} without file stats. Manifest readers can reuse data file instances; use
+   * this method to copy data without stats when collecting files.
+   *
+   * @return a copy of this data file
+   */
+  DataFile slimCopy();
```

Comment:
I don't particularly love the terminology here.  Why not simply add `copy(boolean stats)` or a copy with an enum to indicate what portions of the datafile to include?  At some point we may want just some of the values (e.g. CBO may want counts, but not lower/upper bounds).  Just a thought.

Reply:
I'll change this to `copyWithoutStats` to be more clear.